miasm2/analysis/ssa.py

```diff
@@ -1,6 +1,8 @@
 from collections import deque
 
-from miasm2.analysis.data_flow import DiGraphLivenessIRA, remove_empty_assignblks
+from miasm2.analysis.data_flow import DiGraphLivenessIRA, \
```

Comment:
IMHO, it would be better to have SSA related function in the `ssa.py` file, such as `get_phi_sources_parent_block` and `irblock_has_phi`.
To avoid circular dependency, OutOfSSA can be moved to a separated, dedicated file

Reply:
In fact it depends where you put your "SSA related stuff".
It seems you do the choice here to say that `OutOfSSA` is not linked to `SSA` to avoid this particular circular dependency.
By the way, the `del_unused_edge` depends on `update_phi_with_deleted_edges` which depends on `get_phi_sources_parent_block`...
In fact this may result in the move of a lot of function in the SSA file.
In the current state, SSA contains the bases classes.

Reply:
Yes, and indeed, I also see phi functions in SSA. The idea behind is that SSA could be a kind of self contained file with ssa related stuff, and elements which need ssa will import from this file their necessary stuff. For instance, if I'm looking for the function saying if a block contains a phi or not, I'll probably first have a look at the SSA file. It also avoid putting too many things in data_flow, keeping data flow related stuff.

Reply:
The fact is that today, the SSA is only a *transformation* from one ircfg to another. So the look up argument is somehow not a good one for me.
More over, we will have *a lot* of simplificaiton passes which will operate on the ircfg in SSA form: maybe it won't be a good idea to them in the SSA file neither, I think we agree on this.
As all this is due to the out of ssa, maybe I can set back the SSA with it's minimalist form, let the basic function in it, and put outofssa in the file `outofssa.py` (or `unssa.py`?) which will depend on ssa.
We will have:
- ssa stand alone (class + basic functions)
- outofssa: depends on ssa 
- data_flow: classic dependencies

Does it sound good to you @commial ?

Reply:
Yep, sounds good

Reply:
To me a draw back will the the `DiGraphLivenessSSA` can't be placed in SSA. The solutions are in `unssa.py` or in `data_flow.py`, as the function depends on `ssa.py` AND on `data_flow.py` :cry:

Reply:
Ok, If we put `DiGraphLivenessSSA` in data_flow and remove the `remove_empty_block` in unssa, unssa only depends on SSA.
so:
- ssa self content
- unssa depends on ssa
- data_flow depends on both

Sounds good to me: data_flow does high level stuff, so it's the user of those "low" modules.

Reply:
Perfect, thanks for the refactoring